src/components/routing/routeProtectors/ProfileGuard.js

```diff
@@ -8,17 +8,28 @@ export const ProfileGuard = () => {
 
     useEffect(() => {
         const initiateLogin = async () => {
+            let sendToLogin = false;
+
             setIsLoading(true);
             try {
-                if (!localStorage.getItem("accessToken")) {
-                    const response = await axios.get("http://localhost:8005/loginURL");
-                    const loginURL = response.data;
-                    window.location.href = loginURL;
+                let accessToken = localStorage.getItem("accessToken");
+                if (!accessToken) {
+                    sendToLogin = true;
+                } else {
+                    const response = await axios.get("http://localhost:8005/verifyAccessToken", { params: { accessToken: accessToken } });
+                    if (response.data === false) {
+                        sendToLogin = true;
```

Comment:
It would be better to try to refresh the token rather than sending the user back to the login because it might be that the token is just expired

Reply:
yeah that's right. I'll implement it in the next feature

Reply:
sweet